docs/k0s-install.md

```diff
@@ -0,0 +1,306 @@
+# Install k0s as a service
+
+There are two ways to run k0s binary as a service
+
+1. Using `k0s install` and
+2. Using `k0sctl` tool
+
+## k0s install
+
+`install` command is a helper command for setting up k0s on a brand-new system. It will detect install system init utility installed on the desired host. Supported system inits are systemd and open-rc. NOTE: Windows is currently not supported
+
+Install can be executed for two k0s roles: controller and worker.
+
+Install `controller` subcommand will accept all available flags and parameters supported.
+
+```
+$ k0s install controller --help
+
+Helper command for setting up k0s as controller node on a brand-new system. Must be run as root (or with sudo)
+
+Usage:
+  k0s install controller [flags]
+
+Aliases:
+  controller, server
+
+Examples:
+All default values of controller command will be passed to the service stub unless overriden. 
+
+With controller subcommand you can setup a single node cluster by running:
+
+        k0s install controller --enable-worker
+
+
+Flags:
+  -c, --config string            config file (default: ./k0s.yaml)
+      --cri-socket string        contrainer runtime socket to use, default to internal containerd. Format: [remote|docker]:[path-to-socket]
+  -d, --debug                    Debug logging (default: false)
+      --enable-worker            enable worker (default false)
+  -h, --help                     help for controller
+  -l, --logging stringToString   Logging Levels for the different components (default [konnectivity-server=1,kube-apiserver=1,kube-controller-manager=1,kube-scheduler=1,kubelet=1,kube-proxy=1,etcd=info,containerd=info])
+      --profile string           worker profile to use on the node (default "default")
+      --token-file string        Path to the file containing join-token.
+
+Global Flags:
+      --data-dir string        Data Directory for k0s (default: /var/lib/k0s). DO NOT CHANGE for an existing setup, things will break!
+      --debugListenOn string   Http listenOn for debug pprof handler (default ":6060")
+```
+
+For example runnig a single node k0s cluster on Ubuntu 20.10:
+
+```
+$ k0s install controller --enable-worker
+INFO[2021-02-24 11:05:42] no config file given, using defaults         
+INFO[2021-02-24 11:05:42] creating user: etcd                          
+INFO[2021-02-24 11:05:42] creating user: kube-apiserver                
+INFO[2021-02-24 11:05:42] creating user: konnectivity-server           
+INFO[2021-02-24 11:05:42] creating user: kube-scheduler                
+INFO[2021-02-24 11:05:42] Installing k0s service
+```
+
+Run k0s as a service
+
+```
+$ systemctl start k0scontroller.service
+```
+
+Check service status:
+
+```
+# systemctl status
+● k0s
+    State: running
+     Jobs: 0 queued
+   Failed: 0 units
+    Since: Wed 2021-02-24 10:49:41 UTC; 30min ago
+   CGroup: /
+   ...
+```
+
+Poll cluster for the status:
```

Comment:
```suggestion
Query cluster status:
```

I don't think it polls, I guess it exits with error if the cluster isn't up at that moment.